.github/workflows/build.yml

```diff
@@ -5,14 +5,25 @@ on: [push, pull_request]
 jobs:
   build-macos:
     runs-on: macOS-latest
+    env:
+      CCACHE_COMPRESS: 1
+      CCACHE_TEMPDIR: /tmp/.ccache-temp
     steps:
     - uses: actions/checkout@v1
       with:
         submodules: recursive
+    - name: ccache
+      uses: actions/cache@v1
+      with:
+        path: ~/.ccache
+        key: ccache-macos-build-${{ github.sha }}
+        restore-keys: ccache-macos-build-
     - name: update brew and install dependencies
-      run: brew update && brew install boost hidapi zmq libpgm miniupnpc ldns expat libunwind-headers protobuf
+      run: brew update && brew install boost hidapi zmq libpgm miniupnpc ldns expat libunwind-headers protobuf ccache
     - name: build
-      run: make -j3
+      run: |
+        ccache --max-size=100M
```

Comment:
My humble opinion is, that 100 megabytes for Monero is at least 20 times too little. The cached files will get replaced sooner than they can ever be reused.
My estimated amount comes from the size of the build directory - about 2GB.

Reply:
Travis also uses 100MB cache and it seems to work quite well. I can increase it a bit but Github Actions has a 5GB/repo limit.

Would you share one cache between operating systems? I didn’t do this but would this be possible?

Reply:
Actually I've just realized, that my logic was flawed. The files, which concern ccache are object files ONLY. The large size of the build directory is the result of having so many targets linked statically, which is out of scope for ccache. To measure the real size of files getting into the cache, you may run the command:

`find . -type f -name '*.o' -exec du -ch {} + | grep total`

which yields:
`87M	total`

So 20 times - definitely not, but say 3 times would still be well below Github limits. Are there limits for processing power as well? If so, then we have to find some middle ground anyway.

As to sharing the cache between OSes - I wouldn't do it, since we're getting into a risky territory.